eclair-core/src/test/scala/fr/acinq/eclair/channel/states/ChannelStateTestsHelperMethods.scala

```diff
@@ -228,18 +228,38 @@ trait ChannelStateTestsHelperMethods extends TestKitBase {
     channelUpdateListener.expectMsgType[LocalChannelUpdate]
   }
 
-  def localOrigin(replyTo: ActorRef): Origin.LocalHot = Origin.LocalHot(replyTo, UUID.randomUUID)
+  def localOrigin(replyTo: ActorRef): Origin.LocalHot = Origin.LocalHot(replyTo, UUID.randomUUID())
 
-  def makeCmdAdd(amount: MilliSatoshi, destination: PublicKey, currentBlockHeight: Long, paymentPreimage: ByteVector32 = randomBytes32(), upstream: Upstream = Upstream.Local(UUID.randomUUID), replyTo: ActorRef = TestProbe().ref): (ByteVector32, CMD_ADD_HTLC) = {
+  def makeCmdAdd(amount: MilliSatoshi, destination: PublicKey, currentBlockHeight: Long): (ByteVector32, CMD_ADD_HTLC) = {
+    makeCmdAdd(amount, CltvExpiryDelta(144), destination, randomBytes32(), currentBlockHeight, Upstream.Local(UUID.randomUUID()))
+  }
+
+  def makeCmdAdd(amount: MilliSatoshi, destination: PublicKey, currentBlockHeight: Long, paymentPreimage: ByteVector32): (ByteVector32, CMD_ADD_HTLC) = {
+    makeCmdAdd(amount, destination, currentBlockHeight, paymentPreimage, Upstream.Local(UUID.randomUUID()))
+  }
+
+  def makeCmdAdd(amount: MilliSatoshi, destination: PublicKey, currentBlockHeight: Long, paymentPreimage: ByteVector32, upstream: Upstream): (ByteVector32, CMD_ADD_HTLC) = {
+    makeCmdAdd(amount, CltvExpiryDelta(144), destination, paymentPreimage, currentBlockHeight, upstream)
+  }
+
```

Comment:
Isn't this strictly equivalent to what you're replacing?

Reply:
Yes it's exactly equivalent, the only distinction is that it's nicer to use in tests: you don't have to name arguments and you can put the most relevant arguments at the beginning of the call, so it makes tests slightly easier to read.

> You should add new optional arguments at the end and it won't break anything.

I'd like to avoid that because I'd rather have the important arguments (the one we're actually testing) at the beginning of the arguments list. I agree it's a very minor change, but it's a minor help in test readability, and the tests I was updating in `NormalStateSpec` could use some help to make them easier to read!

Reply:
Since both versions are equivalent, I think we should chose the shorter one. I don't see how expanding one line to nine lines makes it more readable.

```suggestion
  def makeCmdAdd(amount: MilliSatoshi, destination: PublicKey, currentBlockHeight: Long, paymentPreimage: ByteVector32 = randomBytes32(), upstream: Upstream = Upstream.Local(UUID.randomUUID())): (ByteVector32, CMD_ADD_HTLC) = {
    makeCmdAdd(amount, CltvExpiryDelta(144), destination, paymentPreimage, currentBlockHeight, upstream)
  }
```

Reply:
I understand that it is not possible because of limitations from the scala compiler.

Reply:
Yes exactly, unfortunately because we are inside a `trait`...